swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -301,6 +305,7 @@ func runRetrievalTest(chunkCount int, nodeCount int) error {
 		// or until the timeout is reached.
 		allSuccess := false
 		for !allSuccess {
+			allSuccess = true
```

Comment:
forget this allSuccess and localSuccess.
The logic is as simple as this:

```golang
REPEAT:
for {
   for NODES {
        for CHUNKS/FILES {
           if !RETRIEVED {
               // there is some failure on some node, so wait maybe and repeat experiment 
               SLEEP
               continue REPEAT
           }
        }
    }
    // all nodes and files found, exit loop and return without error
    return nil
 }
```

Reply:
Yes, very elegant